pkg/v1/tarball/write.go

```diff
@@ -222,6 +222,10 @@ func writeImagesToTar(refToImage map[name.Reference]v1.Image, m []byte, size int
 func calculateManifest(refToImage map[name.Reference]v1.Image) (m Manifest, err error) {
 	imageToTags := dedupRefToImage(refToImage)
 
+	if len(imageToTags) == 0 {
+		return nil, errors.New("at least one reference must be provided to calculate a manifest")
```

Comment:
I realize this is mostly because we are inconsistent in how we're wrapping errors, and we should do better here, but in the common path we're going to see this returned as:

```
unable to calculate manifest: at least one reference must be provided to calculate a manifest
```

That seems a bit redundant, so I might change this error to `set of images is empty` or something along those lines.